### PR TITLE
bears/ruby: Add RuboCopBear

### PR DESCRIPTION
**Pull request by mr-karan**
This bear uses RuboCop which is a static code
analyzer for Ruby.

Fixes: https://github.com/coala-analyzer/coala-bears/issues/189

### CODE REVIEW

**Comment by sils1297** [comment]
hey @mr-karan if you get the "unknown error" thing in circleci a rebase might solve this

**Comment by sils1297** [comment]
1. You need to implement `check_prerequisites` so the bear checks if rubocop is available and errors out nicely if it's not. This will autoskip your tests on travis and appveyor where we currently don't install gem deps.
2. I filed https://github.com/bbatsov/rubocop/issues/2932 , let's see what they say about that.

**Comment by sils1297** [comment]
f40ed85 needs work

**Comment by sils1297** [comment]
This is also related to coala-analyzer/coala#1804

File: bears/ruby/RuboCopBear.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        r'(?P<col>\d+):\s*(:?(?P<severity>[RCW]|[EF])):\s*'
+        r'(?P<message>.+)')
+    severity_map = {
+        "W": RESULT_SEVERITY.NORMAL,
+        "C": RESULT_SEVERITY.NORMAL,
+        "R": RESULT_SEVERITY.NORMAL,
+        "E": RESULT_SEVERITY.MAJOR,
+        "F": RESULT_SEVERITY.MAJOR
+        }
+
```
Comment (sudheesh001) [inline bears/ruby/RuboCopBear.py]: Align severity in decreasing order `MAJOR => NORMAL => INFO`, It makes things easier.

**Pull request closed without merging**